brmp/fit.py

```diff
@@ -139,6 +141,28 @@ def summary(arr, qs=default_quantiles, row_labels=None):
     col_labels = ['mean', 'sd'] + format_quantiles(qs)
     return ArrReprWrapper(marginal_stats(arr, qs), row_labels, col_labels)
 
+def gelman_rubin(samples):
+    if ((samples.shape[0] < 2 and samples.shape[1] < 4) or
+        (samples.shape[0] >= 2 and samples.shape[1] < 2)):
+        return None # Too few chains or samples.
+    elif samples.shape[0] >= 2:
+        return diags.gelman_rubin(samples)
+    else:
+        return diags.split_gelman_rubin(samples)
+
+def effective_sample_size(samples):
+    if samples.shape[1] < 2:
+        return None # Too few samples.
+    else:
+        return diags.effective_sample_size(samples)
+
+def compute_diag_or_default(diag, samples):
+    val = diag(samples)
+    if val is not None:
+        return val
+    else:
+        return np.zeros((samples.shape[2],))
```

Comment:
Do you think 0 is a good default value, or should we default to a NaN array instead?

Reply:
No, good point, probably not. I've pushed a commit that switches this to nan, though I'm not sure that it improves things much. (When I first saw this in numpyro I though it was a bug, IIRC.) I think this is OK for now though, and I've opened #28 on the possibility of doing something else eventually.